src/core/tests/type_prop/framework_node.cpp

```diff
@@ -50,3 +50,48 @@ TEST(type_prop, framework_node) {
 
     ASSERT_THROW(f_node->validate_and_infer_types(), ov::Exception);
 }
+
+TEST(type_prop, dynamic_framework_node_with_dynamic_input) {
+    auto param = std::make_shared<ov::opset8::Parameter>(ov::element::dynamic, ov::PartialShape::dynamic());
+    auto f_node = std::make_shared<ov::op::util::FrameworkNode>(ov::OutputVector{param});
+
+    // Set partially dynamic shape
+    param->set_partial_shape(ov::PartialShape{ov::Dimension::dynamic(), 64});
+    param->validate_and_infer_types();
+
```

Comment:
Also, you can add testing with determined input type like `f32`, etc.

Reply:
Such test already exist.

Reply:
this is about not static shapes, question is about the element type:)